addons/purchase/models/purchase.py

```diff
@@ -658,9 +658,10 @@ def retrieve_dashboard(self):
                 """
         self._cr.execute(query, (one_week_ago, self.env.company.id))
         res = self.env.cr.fetchone()
-        result['all_avg_order_value'] = round(res[0] or 0, res[3])
         result['all_avg_days_to_purchase'] = round(res[1] or 0, 2)
-        result['all_total_last_7_days'] = round(res[2] or 0, res[3])
+        currency = self.env.company.currency_id
+        result['all_avg_order_value'] = format_amount(self.env, res[0] or 0, currency)
+        result['all_total_last_7_days'] = format_amount(self.env, res[2] or 0, currency)
```

Comment:
Overall looks good to me. Could you just delete the `res[3]` parts of the query since your change will make it unused?

Specifically the `MIN(curr.decimal_places)` and `JOIN res_currency curr ON (comp.currency_id = curr.id)` and the reference to it in the comment above it.

Reply:
@lathuat1997 Please delete the `res[3]` parts of the query to get this merged.

Reply:
@ticodoo @davidtranhp Yes. I deleted it. Thank you!
@ticodoo Can you merge it ?